vignettes/ebola.Rmd

```diff
@@ -0,0 +1,216 @@
+      dplyr::group_by(district) |>
+      dplyr::summarise(cases = dplyr::n())
+  )
+```
+
+(ref:ebola-cloropleth) Figure caption.
+
+```{r ebola-cloropleth, fig.cap="(ref:ebola-cloropleth)"}
+ggplot(sierra_leone_ebola_data_sf, aes(fill = cases, geometry = geometry)) +
+  geom_sf() +
+  scale_fill_viridis_c(na.value = "lightgrey") +
+  theme_minimal() +
+  labs(fill = "Cases")
+```
+
+(ref:ebola-outbreak) Figure caption.
+
+```{r ebola-outbreak, fig.cap="(ref:ebola-outbreak)", fig.height=9}
+# Might want to move to using data.table code rather than dplyr here/everywhere
+sierra_leone_ebola_data |>
+  dplyr::filter(case %% 10 == 0) |>
+  ggplot() +
+    geom_segment(
+      aes(
+        x = date_of_symptom_onset, xend = date_of_sample_tested,
+        y = case, yend = case
+      ),
+      col = "grey"
+    ) +
+    geom_point(aes(x = date_of_symptom_onset, y = case), col = "#56B4E9") +
+    geom_point(aes(x = date_of_sample_tested, y = case), col = "#009E73") +
+    facet_grid(district ~ .) +
+    labs(x = "", y = "Case ID") +
+    theme_minimal()
+```
+
+To prepare the data for use in `epidist`, we first rename the date columns as `ptime` (time of primary event) and `stime` (time of secondary event), as well as transforming to a `data.table`:
+
+```{r}
+sierra_leone_ebola_data <- sierra_leone_ebola_data |>
+  dplyr::mutate(
+    date_of_symptom_onset = lubridate::ymd(date_of_symptom_onset),
+    date_of_sample_tested = lubridate::ymd(date_of_sample_tested),
+    ptime = as.numeric(date_of_symptom_onset - min(date_of_symptom_onset)),
+    stime = as.numeric(date_of_sample_tested - min(date_of_symptom_onset))
+  ) |>
+  dplyr::select(case, ptime, stime, age, sex, district) |>
+  data.table::as.data.table()
+```
+
+Next, we use `observe_process()` to do add interval censoring columns:
+
+```{r}
+obs_cens <- epidist::observe_process(sierra_leone_ebola_data)
+
+all(obs_cens$ptime == obs_cens$ptime_daily)
+all(obs_cens$stime == obs_cens$stime_daily)
+
+max(obs_cens$ptime)
+max(obs_cens$stime)
+```
+
+The columns added are:
+
+* `ptime_daily`: As the `ptime` was already daily then this is superfluous
+* `ptime_lwr`:
+* `ptime_upr`:
+* `stime_daily`: As the `ptime` was already daily then this is superfluous
+* `stime_lwr`:
+* `stime_upr`:
+* `delay_daily`:
+* `delay_lwr`:
+* `delay_upr`:
+* `obs_at`: The maximum value of `stime`
+
+# Fitting district-sex stratified delay distributions
+
+Fit most obvious lognormal with sex effect and district effect.
+Translate fitted distribution to probability mass function
+First fit model with no district-sex stratification.
+Sub-sample the data for speed.
+Use ADVI for speed.
+
+```{r}
+N <- nrow(obs_cens)
+obs_cens <- obs_cens[complete.cases(obs_cens)]
+N_complete <- nrow(obs_cens)
+obs_cens <- obs_cens[sample(seq_len(.N), N_complete / 10, replace = FALSE)]
+obs_prep <- as_latent_individual(obs_cens)
+
+fit <- epidist::epidist(
+  data = obs_prep,
+  formula = brms::bf(mu ~ 1, sigma ~ 1),
```

Comment:
if we rexport bf and lognormal() etc will 1. make it clear which families we have build support for and which may need new tooling and two mean that users don't need to load or specify brms when using them? What do you think?

Reply:
I don't know what rexport is but if it is as it sounds then yes that could be reasonable. There are lots of these `brms::lognormal` similar ones. Also we could do this with `stats::` ones?

Reply:
Created a new issue about this: https://github.com/epinowcast/epidist/issues/254